docs/embed-c.md

```diff
@@ -1,4 +1,112 @@
 # Embedding Wasmtime in C
 
-... more coming soon
+This document shows an example of how to embed Wasmtime using the [C API](https://github.com/WebAssembly/wasm-c-api) to execute a simple wasm program.
 
+## Hello, world!
+
+An excellent example of C API usage can be found at https://github.com/WebAssembly/wasm-c-api/blob/master/example/hello.c
+
+### WebAssembly code
+
+```wat
+(module
+  (func $hello (import "" "hello"))
+  (func (export "run") (call $hello))
+)
+```
+
+### C code
+
+The wasmtime engine and its storage has to be initialized, before the WebAssembly modules can be loaded:
```

Comment:
Nit: When we refer to Wasmtime and we're not talking about the crate name or CLI binary name, we capitalize Wasmtime.